src/components/page/FoodsPage.tsx

```diff
@@ -1,3 +1,156 @@
+  const containerIdMap = createContainerIdNameMap(containers);
+  const groupIdMap = createGroupIdNameMap(containers);
+  const updateUrlParams = (key: string, value?: string | string[]) => {
+    if (!value || (Array.isArray(value) && value.length === 0)) {
+      params.delete(key);
+    } else {
+      params.set(key, Array.isArray(value) ? value.join(',') : value);
+    }
+    replace(`${pathname}?${params.toString()}` as Route);
+  };
+
+  const removeGroupFilter = () => {
+    updateUrlParams('group');
+  };
+
+  const removeContainerFilter = () => {
+    updateUrlParams('container');
+  };
+
+  const removeCategoryFilter = (key: string) => {
+    const updatedCategoryList = categoryList.filter((category) => category !== key);
+    setCategoryList(updatedCategoryList);
+    updateUrlParams('category', updatedCategoryList);
+  };
+
+  useEffect(() => {
+    const categoryList = searchParams?.get('category')?.split(',') || [];
+    setCategoryList(categoryList);
+    setQuery(searchParams?.get('query') || '');
+
+    const filterByGroup = (row: { groupId: string }) => group === '' || group === row.groupId;
+    const filterByContainer = (row: { id: string }) => container === '' || container === row.id;
+    const filterByCategory = (food: { category: string }) => {
+      if (!categoryList.length) return true;
+      return categoryList.some((c) => food.category.includes(c));
+    };
+    const filterByName = (food: { name: string }) =>
+      food.name.toLowerCase().includes(query.toLowerCase());
+
+    const filteredContainers = containers.filter(filterByGroup).filter(filterByContainer);
+    const initialFoodsView: IFoodView[] = filteredContainers.flatMap((container: IContainer) =>
+      container.foods.map((food: IFood) => ({
+        ...food,
+        container: container.name,
+      })),
+    );
+    const filteredFoods = initialFoodsView.filter(filterByName).filter(filterByCategory);
+
+    const sortFoods = (a: IFood, b: IFood) => {
+      switch (sort) {
+        case 'name':
+          return a.name.localeCompare(b.name);
+        case 'expiry':
+          return a.expiry < b.expiry ? -1 : 1;
+        default:
+          return a.createdAt > b.createdAt ? -1 : 1; // default is createdAt desc order
+      }
+    };
+
+    const sortedFoods = [...filteredFoods].sort(sortFoods);
+    setDisplayedFoods(sortedFoods);
+  }, [containers, query, sort, group, container, searchParams]);
+
+  return (
+    <div className="mt-6 mx-4 mb-16">
+      <div className="relative">
+        <SearchBar />
+        <FilterButton
+          containers={containersGroupByGroups}
+          containerIdMap={containerIdMap}
+          groupIdMap={groupIdMap}
+        />
+      </div>
+      <div
+        className={`flex ${
+          group || container || categoryList.length > 0
+            ? 'gap-1.5 mt-4 mb-1.5 sm:overflow-auto sm:whitespace-normal overflow-scroll whitespace-nowrap'
```

Comment:
Could you also replace the margin-bottom with padding-bottom? 

<img width="351" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/22ca195a-f2d5-458c-a6ff-aaf7ab7341e8">

Because the scroll bar overlays the badge.
Screen Recording 2023-12-21 at 19.23.08.mov